.github/workflows/main.yml

```diff
@@ -5,8 +5,14 @@ on: [push, pull_request]
 env:
   DEVELOPER: 1
 
+permissions:
+  contents: read
+
 jobs:
   ci-config:
+    permissions:
+      contents: read
```

Comment:
Is this necessary even if those permissions are already configured globally?

Reply:
According to the documentation permissions are not merged, but replaced. By defining new permission for `actions` we need to repeat the permission for `contents`.